scripts/check_links.py

```diff
@@ -0,0 +1,48 @@
+#! /usr/bin/env python
+"""
+Goes through all the inline-links in markdown files and reports the breakages.
+"""
+import re
+import sys
+import pathlib
+from multiprocessing.dummy import Pool
+from collections import namedtuple
+from typing import Tuple
+import requests
+
+MatchTuple = namedtuple('MatchTuple', 'source name link')
+
+def url_ok(match_tuple: MatchTuple) -> Tuple[MatchTuple, bool]:
+    try:
+        return (match_tuple, requests.get(match_tuple.link).ok)
+    except requests.ConnectionError:
+        return (match_tuple, False)
+
+if __name__ == "__main__":
+
+    project_root = (pathlib.Path(__file__).parent / "..").resolve() # pylint: disable=no-member
+    markdown_files = project_root.glob('**/*.md')
+
+    all_matches = set()
+    url_regex = re.compile(r'\[([^!][^\]]+)\]\((http[s]?[^)(]+)\)')
+    for markdown_file in markdown_files:
+        with open(markdown_file) as handle:
+            for line in handle.readlines():
+                matches = url_regex.findall(line)
+                for match in matches:
```

Comment:
this would be a little bit cleaner if you did

```
for name, link in matches:
```

Reply:
Sure 👍 Thanks